com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ProtoBuilders.java

```diff
@@ -106,7 +107,7 @@ public static class ConvexShapeBuilder extends ProtoBuilder<ConvexShape.Builder>
     public static class CollisionObjectBuilder extends ProtoBuilder<CollisionObjectDesc.Builder> {
 
         private void ValidateShapeTypes(List<Shape> shapeList, IResource resource) throws IOException, CompileExceptionError {
-            String physicsTypeStr = this.project.getProjectProperties().getStringValue("physics", "type", "2D").toUpperCase();
+            String physicsTypeStr = this.project.getProjectProperties().getStringValue("physics", "type", "2D").toUpperCase(Locale.ROOT);
```

Comment:
I think this highlights the issue with the PR, that it's a bit "spotty".
We're "randomly" applying the fix, and there is almost certainly going to be an issue if anyone writes new code, since they're not going to know about this weird and subtle issue.

Can this be set at startup?

Reply:
Good point! Let me check if this is possible.

Reply:
There is Locale.setDefault() which can be used to the locale for the VM. This will affect all locale-sensitive functions (such as toUpperCase).

https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html#setDefault%28java.util.Locale%29

Reply:
Yep, changed this on bob startup instead.